src/lakefs_spec/spec.py

```diff
@@ -528,6 +548,11 @@ def _open(
         -------
         LakeFSFile
             A local file-like object ready to hold data to be received from / sent to a lakeFS server.
+
+        Raises
+        ------
+        NotImplementedError
+            If ``mode`` is not supported.
```

Comment:
I dislike this, because the `Literal` typing already clearly states which values are supported. Of course, it's technically what happens when you pass a bogus mode, but this is too verbose in my opinion.

Reply:
It's an edge case (and I don't really care about it either way, TBH).

The code technically does raise this exception, so documenting it makes sense.
OTOH, the user never interacts with `_open` directly, so they won't even see this docstring in the first place - making the discussion around verbosity kinda moot.

In general, I like having an extra (automated) set of eyes on the docs, since the tool already caught a few places where the docstrings were not in sync with the code. In that sense, for me, the benefits outweigh the edge cases we're discussing here in the PR. (I have yet to see a user complain about documentation being too complete).